lib/hydra/pcdm/deep_member_iterator.rb

```diff
@@ -0,0 +1,25 @@
+module Hydra::PCDM
+  class DeepMemberIterator
+    include Enumerable
+    attr_reader :record
+    def initialize(record)
+      @record = HasMembers.new(record)
+    end
+
+    def each
+      record.members.each do |member|
+        yield member
+      end
+      record.members.each do |member|
+        DeepMemberIterator.new(member).each do |deep_member|
+          yield deep_member
+        end
+      end
+    end
+  end
+  class HasMembers < SimpleDelegator
```

Comment:
Is this a less evil way to monkeypatch?  I dig it.  Name for this is... decoration? façade? :question: